common/rust/cctrusted_base/src/tdx/common.rs

```diff
@@ -0,0 +1,42 @@
+#![allow(non_camel_case_types)]
+use crate::cc_type::*;
+use std::collections::HashMap;
+
+// TDX version ID
+#[derive(Clone, Eq, Hash, PartialEq)]
+pub enum TdxVersion {
+    TDX_1_0,
+    TDX_1_5,
+}
+
+// TDX version ID to version string map
+lazy_static! {
+    pub static ref TDX_VERSION_MAP: HashMap<TdxVersion, String> = {
+        let mut map: HashMap<TdxVersion, String> = HashMap::new();
+        map.insert(TdxVersion::TDX_1_0, "1.0".to_string());
+        map.insert(TdxVersion::TDX_1_5, "1.5".to_string());
+        map
+    };
+}
+
+// TDX version ID to device path string map
+lazy_static! {
+    pub static ref TDX_DEVICE_NODE_MAP: HashMap<TdxVersion, String> = {
+        let mut map: HashMap<TdxVersion, String> = HashMap::new();
+        map.insert(TdxVersion::TDX_1_0, TEE_TDX_1_0_PATH.to_string());
+        map.insert(TdxVersion::TDX_1_5, TEE_TDX_1_5_PATH.to_string());
+        map
+    };
+}
+
+// TDX ioctl operation code to be used for get TDX quote and TD Report
+pub enum TdxOperation {
+    TDX_GET_TD_REPORT = 1,
+    TDX_1_0_GET_QUOTE = 2,
+    TDX_1_5_GET_QUOTE = 4,
```

Comment:
I am not sure why we need distinguish 1_0 get quote and 1_5 get quote?

Reply:
these op codes are used to generate ioctl command for TDX 1.0 and TDX 1.5, see details at https://github.com/hairongchen/cc-trusted-api/blob/rust_sdk_framework/common/rust/cctrusted_base/src/tdx/quote.rs#L110